storage/filesystem/dotgit/dotgit.go

```diff
@@ -1105,10 +1107,24 @@ func (d *DotGit) Module(name string) (billy.Filesystem, error) {
 	return d.fs.Chroot(d.fs.Join(modulePath, name))
 }
 
+func (d *DotGit) AddAlternate(remote string) error {
+	altpath := d.fs.Join(objectsPath, infoPath, alternatesPath)
+	f, err := d.fs.OpenFile(altpath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0640)
+	if err != nil {
+		return err
+	}
+	line := path.Join(remote, objectsPath) + "\n"
+	_, err = io.WriteString(f, line)
+	if err != nil {
+		return fmt.Errorf("error writing 'alternates' file: %w", err)
+	}
+	return f.Close()
```

Comment:
We are not closing the file when the file opens successfully but errors on writing. I am suggesting a `defer f.Close()` to mitigate that.

```suggestion
	return nil
```

Reply:
make sense